scalding-core/src/test/scala/com/twitter/scalding/CoreTest.scala

```diff
@@ -1849,3 +1857,68 @@ class DailySuffixTsvTest extends WordSpec with Matchers {
       .finish
   }
 }
+
+class TestMapper extends MapReduceBase with Mapper[LongWritable, Text, Text, IntWritable] {
+  private val one: IntWritable = new IntWritable(1)
+  private var word: Text = new Text()
+
+  @throws(classOf[IOException])
+  override def map(key: LongWritable,
+    value: Text,
+    output: OutputCollector[Text, IntWritable],
+    reporter: Reporter) = {
+    val line: String = value.toString
+    val itr: StringTokenizer = new StringTokenizer(line)
+    while (itr.hasMoreTokens) {
+      word.set(itr.nextToken)
+      output.collect(word, one)
+    }
+  }
+}
+
+class TestReducer extends MapReduceBase with Reducer[Text, IntWritable, Text, IntWritable] {
+  @throws(classOf[IOException])
+  override def reduce(key: Text,
+    values: java.util.Iterator[IntWritable],
+    output: OutputCollector[Text, IntWritable],
+    reporter: Reporter) = {
+    var sum: Int = 0
+    while (values.hasNext) {
+      sum += values.next.get
+    }
+    output.collect(key, new IntWritable(sum))
+  }
+}
+
+class MapReduceTestJob(args: Args) extends MapReduceJob(args) {
+  override def processJobSpecifics: Unit = {
+    job.setMapOutputKeyClass(classOf[Text])
+    job.setMapOutputValueClass(classOf[LongWritable])
+    job.setOutputKeyClass(classOf[Text])
+    job.setOutputValueClass(classOf[Text])
+    job.setMapperClass(classOf[TestMapper])
+    job.setReducerClass(classOf[TestReducer])
+  }
+}
+
+class MapReduceTestJobTest extends WordSpec with Matchers {
+
+  "A MapReduce Job" should {
+    JobTest(new MapReduceTestJob(_))
+      .arg("mapreducejob.input", "fakeIn")
+      .arg("mapreducejob.output", "fakeOut")
+      .source(TextLine("fakeIn"), List("0" -> "single test", "1" -> "single result"))
```

Comment:
totally blown away that this could work. How is the source plumbed through to the map-reduce job that knows nothing of our test mocking or cascading schemes/etc... Does this really work?